trick_source/sim_services/VariableServer/VariableServerThread_copy_sim_data.cpp

```diff
@@ -92,14 +93,11 @@ int Trick::VariableServerThread::copy_sim_data() {
         }
 
         // Indicate that sim data has been written and is now ready in the buffer_in's of the vars variable list.
-        var_data_staged = true;
-        packets_copied++ ;
+        if (cyclical) {
+            var_data_staged = true;
+            packets_copied++ ;
+        }
 
         pthread_mutex_unlock(&copy_mutex) ;
     }
-
-
-    return (0) ;
```

Comment:
We need this return 0. Double checked the build and we now get a warning from this.

Reply:
Fixed. I'm not sure why I took that out, probably an accident.